### DISCUSSION TITLE
Support for PL/SQL

### DISCUSSION THREAD

**Discussion by vitocuper** [github/securitylab#121] [General] [answered]
Hi!
Is there any intention to add PL/SQL? and if not, how can I add it?
Best regards!

**Comment by jbj** [comment]
@sj, I think you're the right one to answer that.

**Comment by intrigus-lgtm** [comment]
["QL: Object-oriented Queries on Relational Data"](https://drops.dagstuhl.de/opus/volltexte/2016/6096/pdf/LIPIcs-ECOOP-2016-2.pdf) says this:
> As of March 2016, Semmle’s static analysis platform offers about 2500 individual analysesfor eight languages (C/C++/Objective-C, C#, Cobol, Java, JavaScript, PL/SQL, Python,Scala)

Objective-C support has already been removed as far as I know.
Cobol seems to still be supported on lgtm.com, but also deprecated.
I have never seen either PL/SQL or Scala being supported by QL or mentioned on any CodeQL related site I know.
So either this support is not public (and only for paying clients(?)) or has been removed as well.

**Comment by sj** [comment, marked as answer]
CodeQL does not support PL/SQL. Indeed, we did do some work on it in the past, which is why it was listed on the 2016 academic paper. Unfortunately we never got as far as releasing it on LGTM.com (the predecessor of GitHub code scanning) and the work on it has been halted completely. It's not supported for Enterprise use either. 

@p0 commented in detail on what it takes to implement a new language extractor here: https://github.com/github/securitylab/discussions/134#discussioncomment-29203. 

(for completeness: Scala and Objective-C are indeed no longer supported either)
